packages/frontend/src/pages/Recipe.svelte

```diff
@@ -15,21 +15,31 @@ import ContainerConnectionStatusInfo from '../lib/notification/ContainerConnecti
 import { modelsInfo } from '../stores/modelsInfo';
 import { checkContainerConnectionStatus } from '../utils/connectionUtils';
 import { router } from 'tinro';
+import { InferenceType } from '@shared/src/models/IInference';
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
 
 export let recipeId: string;
 
 // The recipe model provided
 $: recipe = $catalog.recipes.find(r => r.id === recipeId);
 $: categories = $catalog.categories;
+
+// model selected to start the recipe
 let selectedModelId: string;
-$: selectedModelId = recipe?.models?.[0] ?? '';
+$: selectedModelId = recipe?.recommended && recipe.recommended.length > 0 ? recipe?.recommended?.[0] : '';
```

Comment:
Can't be shortened ?

Reply:
Not really, because the previous code had a flaw, it was not taking the possibility than there is not models. Let's say we create a recipe without recommended, then we have an error, this line ensure no error will be raised.